### PR TITLE
Add auto close of an inactive session after X minutes

### PR DESCRIPTION
**Pull request by herby2212**
**Changes**
Added a internal timer in the `SessionManager` to check at `1` minute intervals for sessions that are paused longer then `10` minutes. For this a new property got added to the `SessionInfo` called `LastPausedDate` which contains the timestamp the session was last paused.
When a session passes this check it gets closed automatically without any user intervention needed.

The threshold can be configured via the new `InactiveSessionThreshold` property as part of the `ServerConfiguration`. Setting this property to `0` will disable the check and the auto close function.

**Features**
Implements feature [301](https://features.jellyfin.org/posts/301/option-to-kill-a-stream-from-a-user).

**Note**
This PR is a updated version of PR #9714 based on the current master version to resolve merge conflicts.

### CODE REVIEW

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 1]
```diff
@@ -1798,6 +1873,12 @@ public async ValueTask DisposeAsync()
                 _idleTimer = null;
             }
 
+            if(_inactiveTimer is not null)
```
Comment (1337joe) [inline Emby.Server.Implementations/Session/SessionManager.cs:1876]: This should fix the build failure.

```suggestion
            if (_inactiveTimer is not null)
```

**Review by 1337joe — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/Session/SessionManager.cs (hunk 1 above)
Comment (herby2212) [reply Emby.Server.Implementations/Session/SessionManager.cs:1876]: Thanks did make the change locally, but didn't push the change.

**Review by 1337joe — APPROVED** [review]

**Review by cvium — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 2]
```diff
@@ -585,10 +617,53 @@ private async void CheckForIdlePlayback(object state)
...
+        {
+            var pausedSessions = Sessions.Where(i =>
+                    i.NowPlayingItem is not null
+                    && i.PlayState.IsPaused
+                    && i.LastPausedDate is not null)
+                .ToList();
+
+            if (pausedSessions.Count > 0)
+            {
+                var inactiveSessions = pausedSessions.Where(i => (DateTime.UtcNow - i.LastPausedDate).Value.TotalMinutes > _config.Configuration.InactiveSessionThreshold).ToList();
```
[the remark this answers is not in the document]
Comment (herby2212) [reply Emby.Server.Implementations/Session/SessionManager.cs:636]: Thanks I will take a look into this. Yeah I followed the existing structure as I wasn't sure how much difference from existing code is allowed here.

File: Emby.Server.Implementations/Session/SessionManager.cs [hunk 3]
```diff
@@ -585,10 +617,53 @@ private async void CheckForIdlePlayback(object state)
                 playingSessions = Sessions.Where(i => i.NowPlayingItem is not null)
```
[the remark this answers is not in the document]
Comment (cvium) [reply Emby.Server.Implementations/Session/SessionManager.cs:617]: Looks that way

File: Emby.Server.Implementations/Session/SessionManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cvium) [reply Emby.Server.Implementations/Session/SessionManager.cs:636]: Why was this resolved? I don't see any new commits.

[the remark this answers is not in the document]
Comment (herby2212) [reply Emby.Server.Implementations/Session/SessionManager.cs]: > Why was this resolved? I don't see any new commits.

My bad. Was commited, but not pushed.

File: Emby.Server.Implementations/Session/SessionManager.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (herby2212) [reply Emby.Server.Implementations/Session/SessionManager.cs:617]: I would remove it then in a seperate commit where the threshold for idle playback is also made configurable.

**Review by cvium — APPROVED** [review]

**Comment by satmandu** [comment]
@herby2212 This looks really useful! Would you please resolve the merge conflicts?

**Comment by herby2212** [comment]
Merge conflict resolved I hope it gets merged before there is a new one. Else if any informations are needed  for a merge please inform me.

**Comment by Gauvino** [comment]
@herby2212 does the building check failing is normal cause on other pr it work well ?

**Comment by satmandu** [comment]
I built a docker container of this at https://hub.docker.com/layers/satmandu/jellyfin-testing/latest/images/sha256-d78b402901cdffb00d3746f3caf55fad9565e6250cacb64150e26c3372656a89?context=explore for testing... it seems to be working fine for me.

**Comment by Shadowghost** [comment]
Debug build seems to fail. Once this is resolved we should be good to go.

**Comment by herby2212** [comment]
> Debug build seems to fail. Once this is resolved we should be good to go.


Build is fixed. The indentation got wrong through the merge.

**Review by gaurovgiri — APPROVED** [review]
Looks good to me

**Pull request merged**